### PR TITLE
fix #67 add ContentId property

### LINKED ISSUE #67: Add Content-ID property to MailAttachment
Currently setting a Content-ID header for attachments requires creating a MultiMap for the single header, it would be easier to have a setContentId property similar to the other properties.

### PR DESCRIPTION
**Pull request by alexlehm**
fixes #67 Add Content-ID property to MailAttachment

### CODE REVIEW

File: vertx-mail-client/src/main/asciidoc/js/index.adoc [hunk 1]
```diff
@@ -133,13 +133,12 @@ inline and the Content-ID header as "<contentid@domain>"
...
 ----
 var Buffer = require("vertx-js/buffer");
-var MultiMap = require("vertx-js/multi_map");
 var attachment = {
 };
 attachment.contentType = "image/jpeg";
 attachment.data = Buffer.buffer("image data");
 attachment.disposition = "inline";
-attachment.headers = MultiMap.caseInsensitiveMultiMap().add("Content-ID", "<image1@example.com>");
+attachment.contentId = "<image1@example.com>";
```
Comment (cescoffier) [inline vertx-mail-client/src/main/asciidoc/js/index.adoc]: If the contentId must be wrapped in <>, it should be indicated somewhere.

File: vertx-mail-client/src/test/java/io/vertx/ext/mail/MailAttachmentTest.java [hunk 2]
```diff
@@ -35,12 +37,14 @@ public void testConstructor() {
   @Test
   public void testToJson() {
     assertEquals("{}", new MailAttachment().toJson().encode());
-    assertEquals("{\"data\":\"ZGF0YQ==\",\"contentType\":\"text/plain\",\"disposition\":\"inline\",\"description\":\"description\"}",
+    assertEquals("{\"data\":\"ZGF0YQ==\",\"contentType\":\"text/plain\",\"disposition\":\"inline\",\"description\":\"description\",\"contentId\":\"id1@domain.net\",\"headers\":{}}",
```
Comment (cescoffier) [inline vertx-mail-client/src/test/java/io/vertx/ext/mail/MailAttachmentTest.java]: No <> around the contentId. Is it normal (not homogeneous with the documentation)

File: vertx-mail-client/src/test/java/io/vertx/ext/mail/MailAttachmentTest.java [hunk 3]
```diff
@@ -85,7 +97,7 @@ public void testConstructorFromJsonEmpty() {
 
   @Test
   public void testConstructorFromJson() {
-    final String jsonString = "{\"data\":\"YXNkZmc=\",\"name\":\"filename.jpg\"}";
+    final String jsonString = "{\"data\":\"YXNkZmc=\",\"name\":\"filename.jpg\",\"headers\":{\"Header\":[\"Value\"]}}";
```
Comment (cescoffier) [inline vertx-mail-client/src/test/java/io/vertx/ext/mail/MailAttachmentTest.java]: I guess, for compatibility reason, you should also check the the contentId is passed in the headers too.

File: vertx-mail-client/src/main/asciidoc/js/index.adoc (hunk 1 above)
Comment (alexlehm) [inline vertx-mail-client/src/main/asciidoc/js/index.adoc]: changed the adoc accordingly

**Comment by alexlehm** [comment]
changed the test to use a random string in the json test and a strictly formatted content-id in the other test

@cescoffier I am not sure if you mean to test that setting the header sets the contentId property, because that is currently not the case, the headers Map adds to the headers set by the mime encoder since they are added at the end

**Pull request merged**